pydantic/typing.py

```diff
@@ -104,7 +77,8 @@ def get_all_type_hints(obj: Any, globalns: Any = None, localns: Any = None) -> A
 
     def get_origin(t: Type[Any]) -> Optional[Type[Any]]:
         if type(t).__name__ in AnnotatedTypeNames:
-            return cast(Type[Any], Annotated)  # mypy complains about _SpecialForm in py3.6
+            # weirdly this is a runtime requirement, as well as for mypy
```

Comment:
super weird 🤔

Reply:
Ye, things like this scare me a little :fearful:.